sway-core/src/engine_threading.rs

```diff
@@ -42,6 +42,14 @@ impl<'a> Engines<'a> {
             engines: *self,
         }
     }
+
+    /// Helps out some `thing: T` by adding `self` as context.
+    pub fn help_out_with_self<T>(self, thing: T) -> WithEngines<'a, T> {
```

Comment:
I see `help_out()` and this function... is there a better name for it?  'Help out' is a bit ambiguous.

Reply:
I changed it to `with_thing`---`help_out` is from a separate PR. Let me know if you'd prefer something other than `with_thing` 😄

Reply:
I guess!